### ISSUE TITLE
how to use my own rosbag data to train pointpillar model

### ISSUE THREAD

**Issue by fellenB** [open-mmlab/mmdetection3d#916]
I record my liDAR data as a rosbag, however i don't know how to train pointpillar model or test the pretrained model use my own data, can you give me some advices?

**Comment by ZCMax** [comment]
Hi, I recommend you to follow the repo: https://github.com/leofansq/Tools_RosBag2KITTI, converting it to kitti data format. And then you can refer to our customized dataset training tutorial: https://mmdetection3d.readthedocs.io/en/latest/2_new_data_model.html

**Comment by fellenB** [comment]
Thank you very much！i follow  https://github.com/leofansq/Tools_RosBag2KITTI ，converting my rosbag to .bin data. i refer to 2_new_data_model.html, it gives a example for converting waymo data to KITTI data, i wonder how to use the .bin data to test the pretrained model？

**Comment by ZCMax** [comment]
#60

**Comment by ZCMax** [comment]
If you just want to test a pretrained model on you own data, you can organize your .bin data like kitti format, which is shown in https://mmdetection3d.readthedocs.io/en/latest/2_new_data_model.html,  then you need to prepare a config for loading your own data

**Comment by fellenB** [comment]
hv_pointpillars_fpn_sbn-all_4x8_2x_nus-3d.py is used for testing PointPillars on nScenes with 8 gpus,  if i want to test pointPillars on 1 gpu ,how to prepare a config and where can i download the pretrained model?

**Comment by ZCMax** [comment]
1.  ‘4x8’ means 4 samples per GPU using 8 GPUs for training the model, if you want to test on a single gpu, you just need to use tools/test.py,  for multi-gpu testing, you need to use tools/dist_test.sh. For single-gpu and multi-gpu testing, they use the same config file.
2. refer to https://github.com/open-mmlab/mmdetection3d/tree/master/configs/pointpillars,  find nuScenes - fpn - model link.

**Comment by fellenB** [comment]
i test the pretrained model , now i want to save the inference results as a json file use the format below. Is there any interfaces can save json file?

{"name":"000001","timestamp":0,"index":1,"labels":[{"id":3,"category":"vehicle","box3d":{"dimension":{"width":5.07,"length":2.2800000000000002,"height":1.53},"location":{"x":-8.049299170707034,"y":1.9808785162819265,"z":-0.9849999902321764},"orientation":{"rotationYaw":0.01,"rotationPitch":0,"rotationRoll":0}}},{"id":1,"category":"vehicle","box3d":{"dimension":{"width":2.34,"length":4.28,"height":2.05},"location":{"x":-2.4926775298205217,"y":-9.214222951985523,"z":-0.7749999999999999},"orientation":{"rotationYaw":0.22999999999999998,"rotationPitch":0,"rotationRoll":0}}}]}

**Comment by fellenB** [comment]
> If you just want to test a pretrained model on you own data, you can organize your .bin data like kitti format, which is shown in https://mmdetection3d.readthedocs.io/en/latest/2_new_data_model.html, then you need to prepare a config for loading your own data 
Only .bin data needed for testing pretrained model?  Need I to prepare timestamps.txt ,timestamps_start.txt ,timestamps_end.txt files?

**Comment by fellenB** [comment]
[Errno 2] No such file or directory: 'data/kitti/ImageSets/train.txt'

hi, how to creat train.txt?

**Comment by ZCMax** [comment]
> [Errno 2] No such file or directory: 'data/kitti/ImageSets/train.txt'
> 
> hi, how to creat train.txt?

txt under ImageSets are actually the data split, it should be created by yourself, you can just write a script to split your data into train, val , test group. eg: The train.txt includes the id of training data in your whole dataset.

**Comment by ZCMax** [comment]
About training problem, please refer to #429
